### PR TITLE
Migrate DeployableByOLM to Crane

### LINKED ISSUE #184: Migrate Deployable by OLM to Crane
## Bug Description
Migrate the Deployable by OLM check to Crane and increase the unit testing code coverage

## Version and Command Invocation
(The output of `preflight --version`)

## Steps to Reproduce:
(How can we reproduce this?)
1)
2)
3)

## Expected Result
(What did you expect to happen and why?)

## Actual Result
(What actually happened)

## Additional Context
(Anything else you think might help us troubleshoot, like your platform, dependency versions, etc).

### PR DESCRIPTION
**Pull request by samira-barouti**
This PR migrates the DeployableByOLM to Crane. Also refactoring has been done to improve the code readability.

* Previously, the operator data were defined as globals vars. This PR introduces a new type, `OperatorData`, that stores the data that the check needs for the successful deployment of an operator. `OperatorData` is the arg that will be passed to the downstream methods that need these info.
* A new method has been created to fetch operator data from viper and bundle image and return a reference to`OperatorData`. 
* The new OpenshiftEngine has a `*rest.Config` as a field. The check does not pass the `*rest.Config` as a method arg, any longer. 
* All bundle-related constants are moved to default.go

Fixes #184

### CODE REVIEW

File: certification/internal/policy/operator/operator_pkg_name_uniqueness.go [hunk 1]
```diff
@@ -11,14 +11,6 @@ import (
 	log "github.com/sirupsen/logrus"
 )
 
-const (
```
Comment (bcrochet) [inline certification/internal/policy/operator/operator_pkg_name_uniqueness.go:14]: This belongs in a separate PR.

File: certification/internal/k8s/deployable_by_olm_mounted.go [hunk 2]
```diff
@@ -75,7 +75,7 @@ func (p *DeployableByOlmMountedCheck) setUp(bundle string) error {
 	}
 
 	// retrieve the operator metadata from bundle image
-	data, err := containerutil.GetAnnotationsFromBundle(bundle)
```
Comment (bcrochet) [inline certification/internal/k8s/deployable_by_olm_mounted.go:78]: I wouldn't bother with these changes, as this code is going to go away very soon. I'd rather not see it in the same commit.

**Review by bcrochet — CHANGES_REQUESTED** [review]
Couple of comments.

File: certification/internal/policy/operator/operator_pkg_name_uniqueness.go (hunk 1 above)
Comment (samira-barouti) [reply certification/internal/policy/operator/operator_pkg_name_uniqueness.go:14]: The DeployableByOLM requires the package key, among the others. The PackageUniqueness check also requires this key. I was not comfortable using a const which has been defined in another file, and that was the main motivation to move all these common constants into a central place.

File: certification/internal/k8s/deployable_by_olm_mounted.go (hunk 2 above)
Comment (samira-barouti) [reply certification/internal/k8s/deployable_by_olm_mounted.go:78]: sure, I can revert this change.

File: certification/internal/policy/operator/operator_pkg_name_uniqueness.go (hunk 1 above)
Comment (bcrochet) [reply certification/internal/policy/operator/operator_pkg_name_uniqueness.go:14]: Ok. Perfect. I didn't gleam immediately that it was related. Totally fine with that.

**Review by bcrochet — APPROVED** [review]
/lgtm
